src/data/fintype/basic.lean

```diff
@@ -641,6 +641,13 @@ by simp [to_finset]
 @[simp] theorem mem_to_finset_val {s : set α} [fintype s] {a : α} : a ∈ s.to_finset.1 ↔ a ∈ s :=
 mem_to_finset
 
+/-- Membership of a set with a `fintype` instance is decidable.
+
+Using this as an instance leads to potential loops with `subtype.fintype` under certain decidability
+assumptions, so it should only be declared a local instance. -/
+def decidable_mem_of_fintype [decidable_eq α] (s : set α) [fintype s] (a) : decidable (a ∈ s) :=
```

Comment:
Do we want to make this a reducible non-instance?

Reply:
It looks like there aren't any cases of reducible non-instances for `decidable` -- maybe it's not needed? I don't mind adding it if you think it would be useful.

Reply:
Fair enough, I don't see any particular need for it so let's leave it